### ISSUE TITLE
Why is docker_service_manager_upstart restarting docker service (delayed)?

### ISSUE THREAD

**Issue by paustin01** [chef-cookbooks/docker#695]
I'm noticing that on initial provision using the docker_service resource detailed below that docker_service_manager_upstart is sending a delayed `restart` action to docker_service.  Unfortunately this stops all the containers I had set up later in the chef run.  I don't recall seeing this behavior previously but I can't be sure.  Running v2.6.6 of the cookbook.  Any ideas?

```
docker_service 'default' do
  install_method 'package'
  storage_driver 'aufs'
  version '1.10.2'
  tls false
  bip '172.17.0.1/16'
  action [:create, :start]
end
```

**Comment by chasebolt** [comment]
can you post a stack trace? would like to see the reasoning on why it thinks it needs to restart the service,

**Comment by paustin01** [comment]
Looks like the `/etc/default/docker` template is triggering a restart.
```
     * template[/etc/default/docker] action create[2016-04-19T04:08:00+00:00] INFO: Processing template[/etc/default/docker] action create (/tmp/kitchen/cookbooks/docker/libraries/docker_service_manager_upstart.rb line 25)
[2016-04-19T04:08:00+00:00] DEBUG: Providers for generic template resource enabled on node include: [Chef::Provider::Template]
[2016-04-19T04:08:00+00:00] DEBUG: Provider for action create on resource template[/etc/default/docker] is Chef::Provider::Template
[2016-04-19T04:08:00+00:00] DEBUG: template[/etc/default/docker] checksumming file at /etc/default/docker.
[2016-04-19T04:08:00+00:00] DEBUG: calculating checksum of /etc/default/.chef-docker20160419-1654-1lrlgt5 to compare with b395bbf506dec81df45ca68710c190291f5f3ee18d859ae078cbd11acfc5d914
[2016-04-19T04:08:00+00:00] DEBUG: Running: diff -u /etc/default/docker /etc/default/.chef-docker20160419-1654-1lrlgt5
[2016-04-19T04:08:00+00:00] INFO: template[/etc/default/docker] backed up to /tmp/kitchen/backup/etc/default/docker.chef-20160419040800.998845
[2016-04-19T04:08:00+00:00] DEBUG: Reading modes from /etc/default/docker file
[2016-04-19T04:08:00+00:00] DEBUG: Applying mode = 644, uid = 0, gid = 0 to /etc/default/.chef-docker20160419-1654-1lrlgt5
[2016-04-19T04:08:00+00:00] DEBUG: Moving temporary file /etc/default/.chef-docker20160419-1654-1lrlgt5 into place at /etc/default/docker
[2016-04-19T04:08:00+00:00] INFO: template[/etc/default/docker] updated file contents /etc/default/docker

- update content in file /etc/default/docker from b395bb to f77641
--- /etc/default/docker 2016-02-10 20:54:43.000000000 +0000
+++ /etc/default/.chef-docker20160419-1654-1lrlgt5  2016-04-19 04:08:00.992967064 +0000
@@ -1,21 +1,13 @@
 # Docker Upstart and SysVinit configuration file

-#
-# THIS FILE DOES NOT APPLY TO SYSTEMD
-#
-#   Please see the documentation for "systemd drop-ins":
-#   https://docs.docker.com/engine/articles/systemd/
-#
-
 # Customize location of Docker binary (especially for development testing).
-#DOCKER="/usr/local/bin/docker"
+DOCKER="/usr/bin/docker"

 # Use DOCKER_OPTS to modify the daemon startup options.
-#DOCKER_OPTS="--dns 8.8.8.8 --dns 8.8.4.4"
+DOCKER_OPTS="--bip=172.17.0.1/16 --pidfile=/var/run/docker.pid --storage-driver=aufs --tls=false"

 # If you need Docker to use an HTTP proxy, it can also be specified here.
-#export http_proxy="http://127.0.0.1:3128/"

+
 # This is also a handy place to tweak where Docker's temporary files go.
-#export TMPDIR="/mnt/bigdrive/docker-tmp"[2016-04-19T04:08:01+00:00] DEBUG: Found target_mode == nil, so no mode was specified on resource, not managing mode
[2016-04-19T04:08:01+00:00] DEBUG: Found target_uid == nil, so no owner was specified on resource, not managing owner
[2016-04-19T04:08:01+00:00] DEBUG: Found target_gid == nil, so no group was specified on resource, not managing group
[2016-04-19T04:08:01+00:00] DEBUG: selinux utilities can not be found. Skipping selinux permission fixup.

     * service[docker] action start[2016-04-19T04:08:01+00:00] INFO: Processing service[docker] action start (/tmp/kitchen/cookbooks/docker/libraries/docker_service_manager_upstart.rb line 36)
[2016-04-19T04:08:01+00:00] DEBUG: service[docker] found /etc/init/docker.conf
[2016-04-19T04:08:01+00:00] DEBUG: service[docker] enabled: start on (filesystem and net-device-up IFACE!=lo)
[2016-04-19T04:08:01+00:00] DEBUG: service[docker] already running - nothing to do
(up to date)

[2016-04-19T04:08:01+00:00] INFO: template[/etc/default/docker] sending restart action to docker_service_manager_upstart[default] (delayed)
   * docker_service_manager_upstart[default] action restart[2016-04-19T04:08:01+00:00] INFO: Processing docker_service_manager_upstart[default] action restart (/tmp/kitchen/cookbooks/docker/libraries/docker_service.rb line 74)
```
Then the restart
```
[2016-04-19T04:08:01+00:00] INFO: docker_installation_package[default] sending restart action to docker_service[default] (delayed)
         * docker_service[default] action restart[2016-04-19T04:08:01+00:00] INFO: Processing docker_service[default] action restart (openam::test line 15)
```

For reference, this is the entire recipe I was testing with.
```
apt_repository 'docker' do
  uri 'https://apt.dockerproject.org/repo'
  components ['main']
  distribution 'ubuntu-trusty'
  key '58118E89F3A912897C070ADBF76221572C52609D'
  keyserver 'p80.pool.sks-keyservers.net'
  action :add
end

docker_service 'default' do
  install_method 'package'
  storage_driver 'aufs'
  version '1.10.2'
  tls false
  bip '172.17.0.1/16'
  action [:create, :start]
end

log "**************** End of recipe."
```
Let me know if you want to see anything specific.  Thanks Chase!

**Comment by chasebolt** [comment]
thanks that stack trace is helpful. what version are you running for your os, chef, docker cookbook?

**Comment by paustin01** [comment]
Ubuntu 14.04 (vagrant by way of test kitchen), chef version 12.9.38, docker cookbook version 2.6.6.  Thanks Chase.

**Comment by chasebolt** [comment]
fix released in v2.6.8
